main.py

```diff
@@ -16,44 +131,53 @@ def read_send(sock):
             # > however high performance applications will want to
             # > match this parameter to the blocksize parameter used
             # > when opening the stream.
-            samples = b""
+            samples = b''
             while len(samples) < buffer_size:
                 samples += snd.read_from_device(snd.instream.blocksize)
 
             sent = 0
             while sent < len(samples):
                 sent += sock.send(samples[sent:])
 
-            logger.root_logger.debug(f"Read and sent {len(samples)} bytes")
+            logger.root_logger.debug(f"Sound. Read and sent {len(samples)} bytes")
+            time.sleep(0)
+
     except (BrokenPipeError, ConnectionResetError) as e:
         logger.root_logger.warning(e)
 
 
 def receive_play(sock):
     try:
         while True:
-            samples = b""
+            samples = b''
             while len(samples) < buffer_size:
                 samples += sock.recv(buffer_size - len(samples))
 
             snd.write_to_device(samples)
 
             logger.root_logger.debug(
-                f"Received and played {len(samples)} bytes")
+                f"Sound. Received and played {len(samples)} bytes")
+            time.sleep(0)
 
     except (BrokenPipeError, ConnectionResetError) as e:
         logger.root_logger.warning(e)
 
 
-def start_join_threads(sock):
-    read_send_thread = Thread(target=read_send, args=(sock, ))
-    receive_play_thread = Thread(target=receive_play, args=(sock, ))
+def start_join_threads(sock_tcp, sock_udp, pack=b''):
+    read_send_thread = Thread(target=read_send, args=(sock_tcp,))
+    receive_play_thread = Thread(target=receive_play, args=(sock_tcp,))
+    get_and_send_thread = Thread(target=get_and_send_data, args=(sock_udp,))
+    receive_and_play_thread = Thread(target=receive_data, args=(sock_udp, pack,))
 
     read_send_thread.start()
     receive_play_thread.start()
+    get_and_send_thread.start()
+    receive_and_play_thread.start()
```

Comment:
This gets a bit messy. `receive_play()` and `receive_and_play()` are just too similar to be distinguishable.

I think you should refactor this.